File: server/java-jaxrs/src/main/java/com/wordnik/api/EventsApi.java
package com.wordnik.api;

import com.wordnik.swagger.annotations.*;

import com.wordnik.model.Event;
import com.wordnik.model.EventStatus;
import java.util.List;
import com.wordnik.api.NotFoundException;

import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Path("/events")
@Api(value = "/events", description = "the events API")
@Produces({"application/json"})
public class EventsApi {
  @POST
  @Path("/")
  @ApiOperation(value = "adds a new event", notes = "", response = Event.class)
  @ApiResponses(value = { })
     
  public Response addEvent(
    @ApiParam(value = "event to add"
    ,required=true) Event body
    )
      throws NotFoundException {
    // throw 400 error on any null, required values
    if(body == null) {
      return Response.status(400).entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "body is a required parameter")).build();
    }
    // do some magic!
    
    return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "OK!")).build();
  }

  @GET
  @Path("/{eventId}")
  @ApiOperation(value = "looks up an event by ID", notes = "", response = Event.class)
  @ApiResponses(value = { @ApiResponse(code = 404, message = "Event not found")})
     
  public Response findEventById(
    @ApiParam(value = "ID of event to look up"
    ,required=true)@PathParam("eventId") Long eventId
    )
      throws NotFoundException {
    // throw 400 error on any null, required values
    if(eventId == null) {
      return Response.status(400).entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "eventId is a required parameter")).build();
    }
    // do some magic!
    
    return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "OK!")).build();
  }

  @DELETE
  @Path("/{eventId}")
  @ApiOperation(value = "Removes an event", notes = "", response = Event.class)
  @ApiResponses(value = { @ApiResponse(code = 404, message = "Event not found")})
     
  public Response deleteEvent(
    @ApiParam(value = "ID of event to look delete"
    ,required=true)@PathParam("eventId") Long eventId
    )
      throws NotFoundException {
    // throw 400 error on any null, required values
    if(eventId == null) {
      return Response.status(400).entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "eventId is a required parameter")).build();
    }
    // do some magic!
    
    return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "OK!")).build();
  }

  @GET
  @Path("/{eventId}/status")
  @ApiOperation(value = "gets the attendee status of the event", notes = "", response = EventStatus.class)
  @ApiResponses(value = { })
     
  public Response getEventStatus(
    @ApiParam(value = "ID of event to get status on"
    ,required=true)@PathParam("eventId") Long eventId
    )
      throws NotFoundException {
    // throw 400 error on any null, required values
    if(eventId == null) {
      return Response.status(400).entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "eventId is a required parameter")).build();
    }
    // do some magic!
    
    return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "OK!")).build();
  }

  }

File: server/nodejs/App/apis/EventsApi.js
var swagger = require("swagger-node-express");
var url = require("url");
var errors = swagger.errors;
var params = swagger.params;

/* add model includes */

function writeResponse (response, data) {
  response.header('Access-Control-Allow-Origin', "*");
  response.header("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT");
  response.header("Access-Control-Allow-Headers", "Content-Type");
  response.header("Content-Type", "application/json; charset=utf-8");
  response.send(JSON.stringify(data));
}

exports.models = models = require("../models.js");

exports.addEvent = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "/events",
    "notes" : "",
    "summary" : "adds a new event",
    "httpMethod": "POST",
    "params" : [].concat([]).concat([]).concat([params.body("body", "event to add", "Event", true)
    ]),
    "responseClass" : "Event",
    "errorResponses" : [errors.invalid('id'), errors.notFound('Event')],
    "nickname" : "addEvent"
  },
  'action': function (req,res) {
    if (!req.params.body) {
      throw errors.invalid('body');
    }
    writeResponse(res, {message: "how about implementing addEvent as a POST method?"});    
  }
};
exports.findEventById = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "/events/{eventId}",
    "notes" : "",
    "summary" : "looks up an event by ID",
    "httpMethod": "GET",
    "params" : [].concat([params.path("eventId", "ID of event to look up", "long", null, true)]).concat([]).concat([]),
    "responseClass" : "Event",
    "errorResponses" : [errors.invalid('id'), errors.notFound('Event')],
    "nickname" : "findEventById"
  },
  'action': function (req,res) {
    if (!req.params.eventId) {
      throw errors.invalid('eventId');
    }
    writeResponse(res, {message: "how about implementing findEventById as a GET method?"});    
  }
};
exports.deleteEvent = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "/events/{eventId}",
    "notes" : "",
    "summary" : "Removes an event",
    "httpMethod": "DELETE",
    "params" : [].concat([params.path("eventId", "ID of event to look delete", "long", null, true)]).concat([]).concat([]),
    "responseClass" : "Event",
    "errorResponses" : [errors.invalid('id'), errors.notFound('Event')],
    "nickname" : "deleteEvent"
  },
  'action': function (req,res) {
    if (!req.params.eventId) {
      throw errors.invalid('eventId');
    }
    writeResponse(res, {message: "how about implementing deleteEvent as a DELETE method?"});    
  }
};
exports.getEventStatus = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "/events/{eventId}/status",
    "notes" : "",
    "summary" : "gets the attendee status of the event",
    "httpMethod": "GET",
    "params" : [].concat([params.path("eventId", "ID of event to get status on", "long", null, true)]).concat([]).concat([]),
    "responseClass" : "EventStatus",
    "errorResponses" : [errors.invalid('id'), errors.notFound('EventStatus')],
    "nickname" : "getEventStatus"
  },
  'action': function (req,res) {
    if (!req.params.eventId) {
      throw errors.invalid('eventId');
    }
    writeResponse(res, {message: "how about implementing getEventStatus as a GET method?"});    
  }
};

